### PR TITLE
Add more stateless components

### PR DESCRIPTION
**Pull request by neilff**
- Added building block components in /components/ui
- Added login gated modal
- Added about page to provide two routes to navigate between
- Broke out utility functions into separate files for easier reuse
- Added redux form
- Refactored reducers to not use redux-actions
- Renamed user reducer -> session reducer, now has user profile, and a fake session ID

### CODE REVIEW

File: src/reducers/session.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    }));
+
+  case LOGIN_USER_ERROR:
+    return state.merge(fromJS({
+      hasError: true,
+      isLoading: false,
+    }));
+
+  case LOGOUT_USER:
+    return state.merge(fromJS({
```
Comment (andrewdamelio) [inline src/reducers/session.js]: Any reason why not the below?
```js
  case LOGOUT_USER:
    return state.merge(INITIAL_STATE);
```

Comment (neilff) [inline src/reducers/session.js]: Ah good find, will change

**Comment by andrewdamelio** [comment]
@neilff I really enjoy what you've done here with UI components. I believe this is an important concept which I've personally struggled with in projects. Seeing component design already being embrace, within the Seed repo does add a lot of extra components, but I feel like it does provide much needed clarity and examples around component design.

**Comment by SethDavenport** [comment]
:+1:.  I like it

**Pull request merged**